### PR TITLE
[XPU] upgrade sglang xpu backend to PyTorch 2.13

### PR DESCRIPTION
**Pull request by yuchengliu1**
update pytorch to 2.13 for xpu and update other relation. The oneAPI version should be 2026.0 after updated.

## Tests

python3 run_suite.py --hw xpu --suite stage-a-test-1-gpu-xpu

and

python3 run_suite.py --hw xpu --suite stage-b-test-1-gpu-xpu

---
### CI States

Latest PR Test (Base): :x: [Run #31835244191](https://github.com/sgl-project/sglang/actions/runs/31835244191)
Latest PR Test (Extra): :x: [Run #31835244246](https://github.com/sgl-project/sglang/actions/runs/31835244246)

### CODE REVIEW

**Comment by polisettyvarma** [comment]
we already have PR https://github.com/sgl-project/sglang/pull/31031 for pytorch upgrade on XPU
@yuchengliu1 let's use older PR

**Comment by arathi-hlab** [comment]
Ran the PR's stage-a and stage-b suites on Intel Arc Pro B60 Graphics with the PR's Dockerfile (`intel/deep-learning-essentials:2026.0.0-devel-ubuntu24.04`, `torch==2.13.0+xpu`, `sgl-kernel-xpu@main`).

**Stage-A:** 4/4 passed (96.6s).

**Stage-B:** 9/11 passed (708.8s, with `--continue-on-error`). The two failures both look like `torch 2.13.0+xpu` regressions rather than PR bugs:

**1. `test_topk.py::TestBiasedGroupedTopK::test_biased_grouped_topk` — SIGSEGV in `torch.topk` on XPU.**
Reduces to 3 lines of pure PyTorch:
```python
import torch
x = torch.randn(1024, 32, device="xpu")
x.topk(2, dim=-1)   # segfault
```
Full error:
```
Fatal Python error: Segmentation fault
Current thread 0x000073cb99bee740 (most recent call first):
  File "<stdin>", line 3 in <module>
bash: 58516 Segmentation fault (core dumped) python3 ...
exit=139   # 128 + SIGSEGV(11)
```
Controls that confirm the fault is `topk`-on-XPU-specific: same tensor on CPU → `x.topk(2, dim=-1)` returns fine; same XPU tensor with `x + 1` returns fine; a tensor-creation + `torch.xpu.synchronize()` before the topk also succeeds — the crash is inside `.topk()`. No C++ traceback (even with `TORCH_SHOW_CPP_STACKTRACES=1`) — crash happens inside the SYCL kernel before any handler runs. Same crash with `TORCHDYNAMO_DISABLE=1` and independent of `torch.use_deterministic_algorithms`. In sglang the segv originates at `python/sglang/srt/layers/moe/topk.py:1304` (`scores_for_choice.view(...).topk(2, dim=-1)[0]`).

**2. `test_deepseek_ocr.py` — device-side assert in `torch-xpu-ops`.**
During the first prefill after server startup:
```
AssertHandler::printMessage
/__w/pytorch/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/Indexing.cpp:436:
  operator(): global id: [0,0,0], local id: [0,0,0]
  Assertion \`totalElements <= srcSize_\` failed.
Fatal Python error: Aborted
```
Backtrace top:
```
File "torch/xpu/streams.py", line 171 in synchronize
File "sglang/srt/managers/scheduler_components/batch_result_processor.py", line 203 in process_batch_result_prefill
File "sglang/srt/managers/scheduler.py", line 3918 in process_batch_result
```
Scheduler subprocess `SIGABRT`s (exit -6) and surfaces to the client as `Server process exited with code -9`. Not OOM — KV cache allocated cleanly beforehand (`avail mem=4.45 GB`, KV K=5.95 GB / V=5.95 GB, `#tokens=207744`).

Both failures point at upstream `torch-xpu-ops` kernels (topk + indexing). Full logs available on request; happy to file both upstream if useful.

File: test/registered/xpu/test_deepseek_ocr.py [hunk 1]
```diff
@@ -23,8 +23,11 @@
     popen_launch_server,
 )
 
-register_xpu_ci(est_time=360, suite="stage-b-test-1-gpu-xpu")
-
+register_xpu_ci(
+    est_time=360,
+    suite="stage-b-test-1-gpu-xpu",
+    disabled="Temporarily disabled",
+)
```
Comment (arathi-hlab) [inline test/registered/xpu/test_deepseek_ocr.py]: please mentioned disable reason with pt2.13

Comment (mingfeima) [reply test/registered/xpu/test_deepseek_ocr.py]: are we able to reproduce this issue via pure torch? i mean without sglang.

Comment (arathi-hlab) [reply test/registered/xpu/test_deepseek_ocr.py]: @mingfeima  can you reply  @yuchengliu1 ?

**Comment by YangKai0616** [comment]
Also, for torch XPU, using the XPU driver specified in this PR after version 2.13 may cause `torch.nonzero` and `torch.unique` to return truncated or empty results, leading to incorrect results in dynamic-shape operations such as boolean indexing. The workaround can be found in the related issue https://github.com/intel/torch-xpu-ops/issues/4396. Should we update the version accordingly?

Reproduction script：
```
#!/usr/bin/env python3
"""Minimal repro for incorrect dynamic-shape results on PyTorch XPU."""

import platform
import sys

import torch


def summarize(tensor):
    flat = tensor.reshape(-1)
    preview = flat[:16].tolist()
    suffix = "..." if flat.numel() > len(preview) else ""
    return f"shape={tuple(tensor.shape)}, numel={tensor.numel()}, values={preview}{suffix}"


def report(name, operation, expected):
    try:
        actual = operation()
        torch.xpu.synchronize()
        actual_cpu = actual.cpu()
        passed = torch.equal(actual_cpu, expected)
        print(
            f"{name}: {'PASS' if passed else 'FAIL'}\n"
            f"  expected {summarize(expected)}\n"
            f"  actual   {summarize(actual_cpu)}"
        )
        return passed
    except Exception as error:
        print(f"{name}: EXCEPTION\n  {type(error).__name__}: {error}")
        return False


def main():
    print(f"Python: {sys.version.replace(chr(10), ' ')}")
    print(f"Platform: {platform.platform()}")
    print(f"PyTorch: {torch.__version__}")
    print(f"PyTorch git version: {torch.version.git_version}")
    print(f"XPU available: {torch.xpu.is_available()}")

    if not torch.xpu.is_available():
        print("XPU is not available; cannot run the reproducer.")
        return 2

    print(f"XPU device: {torch.xpu.get_device_name(0)}")
    print()

    ids_cpu = torch.arange(282, dtype=torch.int64) % 256
    mask_cpu = torch.zeros(282, dtype=torch.bool)
    mask_cpu[:3] = True
    matrix_cpu = torch.arange(282 * 4, dtype=torch.int64).reshape(282, 4)

    ids_xpu = ids_cpu.to("xpu")
    mask_xpu = mask_cpu.to("xpu")
    matrix_xpu = matrix_cpu.to("xpu")
    torch.xpu.synchronize()

    results = [
        report("torch.unique", lambda: torch.unique(ids_xpu), torch.unique(ids_cpu)),
        report("torch.nonzero", lambda: torch.nonzero(mask_xpu), torch.nonzero(mask_cpu)),
        report("boolean indexing", lambda: matrix_xpu[mask_xpu], matrix_cpu[mask_cpu]),
    ]

    print()
    if all(results):
        print("All XPU results match CPU results.")
        return 0

    print("Reproduced: at least one XPU dynamic-shape operation differs from CPU.")
    return 1


if __name__ == "__main__":
    raise SystemExit(main())
```

Outputs:
```
Python: 3.12.3 (main, Jun 19 2026, 12:46:00) [GCC 13.3.0]
Platform: Linux-6.17.0-41-generic-x86_64-with-glibc2.39
PyTorch: 2.13.0+xpu
PyTorch git version: cf30153c4c131c8164ee7798e5022d810682e2cb
XPU available: True
XPU device: Intel(R) Arc(TM) Pro B60 Graphics

torch.unique: FAIL
  expected shape=(256,), numel=256, values=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]...
  actual   shape=(0,), numel=0, values=[]
torch.nonzero: FAIL
  expected shape=(3, 1), numel=3, values=[0, 1, 2]
  actual   shape=(0, 1), numel=0, values=[]
boolean indexing: FAIL
  expected shape=(3, 4), numel=12, values=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
  actual   shape=(0, 4), numel=0, values=[]

Reproduced: at least one XPU dynamic-shape operation differs from CPU.
```

**Comment by yuchengliu1** [comment]
The error of torch.nonzero and torch.unique can be solved by updated XPU driver.

**Review by arathi-hlab — APPROVED** [review]

File: test/registered/xpu/test_deepseek_ocr.py (hunk 1 above)
Comment (yuchengliu1) [reply test/registered/xpu/test_deepseek_ocr.py]: This failure fixed by update XPU driver
